src/scripts/components/layer/wv.layer.list.js

```diff
@@ -0,0 +1,120 @@
+      minWidth: this.state.width
+    });
+
+    this._rowRenderer = this._rowRenderer.bind(this);
+    this._setListRef = this._setListRef.bind(this);
+    this.saveExpandedInfoState = this.saveExpandedInfoState.bind(this);
+  }
+  componentWillUpdate(){
+    this._cache.clearAll();
+  }
+  /*
+   * Saves the visibility/state of the metadata for each layer
+   * @method saveExpandedInfoState
+   * @param {string} layerID - the layer which has metadata that
+   *  needs to be toggled
+   * @return {void}
+   */
+  saveExpandedInfoState(layer){
+    var { infoExpanded } = this.state;
+    var index = infoExpanded.indexOf(layer);
+    if(index > -1){
+      infoExpanded.splice(index, 1); // Removes layer from expanded list
+    } else {
+      infoExpanded.push(layer);
+    }
+    this.setState({ infoExpanded: infoExpanded });
+  }
+  /*
+   * Recalculates the row height for a given rowIndex
+   * @method recalculateRowHeight
+   * @param {number} index - Index of the row to be recalculated
+   * @return {void}
+   */
+  recalculateRowHeight (rowIndex){
+    this._cache.clear(rowIndex, 0);
+    this._layerList.recomputeRowHeights(rowIndex);
+    this._layerList.scrollToRow(rowIndex);
+  }
+  _rowRenderer ({ index, isScrolling, key, parent, style }) {
+    var enabled = false;
+    for(var i of this.props.model.active){
+      if(i.id === this.state.layerFilter[index])
+        enabled = true;
+    }
+    var current = this.state.layerFilter[index];
+    var expanded = false;
+    if(this.state.infoExpanded.includes(current))
+      expanded = true;
+    style.paddingTop = '5px';  //'Margin' for each list element
+    return (
+      <CellMeasurer
+        cache={this._cache}
+        columnIndex={0}
+        key={key}
+        parent={parent}
+        rowIndex={index}
+      >
+        {({ measure }) => (
+          <LayerRadio
+            key={'layer-'+ current + '-' + key}
+            layerId={current}
+            title={this.props.config.layers[current].title}
+            subtitle={this.props.config.layers[current].subtitle}
+            enabled={enabled}
+            metadata={this.props.metadata[current] || null}
+            expand={this.saveExpandedInfoState}
+            expanded={expanded}
+            style={style}
+            rowIndex={index}
+            onState={this.props.model.add}
+            offState={this.props.model.remove}
+            onChange={this.recalculateRowHeight.bind(this)}
+            onLoad={measure}
+          />
+        )}
+    </CellMeasurer>
+    );
+  }
+  render() {
+    return(
+        <List
+            deferredMeasurementCache={this._cache}
+            id="flat-layer-list"
+            width={this.state.width}
+            height={this.state.height}
+            overscanRowCount={5}
+            ref={this._setListRef}
+            rowCount={this.state.layerFilter.length}
+            rowHeight={this._cache.rowHeight}
+            scrollToAlignment="auto"
+            rowRenderer={this._rowRenderer}
```

Comment:
What do you think about using `(row)=>this._rowRenderer(row)` here and then we can remove the separate `bind()` lines above?